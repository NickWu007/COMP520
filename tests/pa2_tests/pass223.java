// PA2 pass AST multiple classes
class A {}

class B {
    private int [] v;  

    public void foo(int a, B other) {}

    C c;

    private int x;
}

class C {
    private boolean b;

    public int[] tryit() {
        int x =-/* unary */-x;
        return (new int [20]);  
    }
}
