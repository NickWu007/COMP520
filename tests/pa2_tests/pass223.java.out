======= AST Display =========================
Package
  ClassDeclList [3]
  . ClassDecl
  .   "A" classname
  .   FieldDeclList [0]
  .   MethodDeclList [0]
  . ClassDecl
  .   "B" classname
  .   FieldDeclList [3]
  .   . (private) FieldDecl
  .   .   ArrayType
  .   .     INT BaseType
  .   .   "v" fieldname
  .   . (public) FieldDecl
  .   .   ClassType
  .   .     "C" Identifier
  .   .   "c" fieldname
  .   . (private) FieldDecl
  .   .   INT BaseType
  .   .   "x" fieldname
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   VOID BaseType
  .   .   "foo" methodname
  .   .   ParameterDeclList [2]
  .   .   . ParameterDecl
  .   .   .   INT BaseType
  .   .   .   "a"parametername 
  .   .   . ParameterDecl
  .   .   .   ClassType
  .   .   .     "B" Identifier
  .   .   .   "other"parametername 
  .   .   StmtList [0]
  . ClassDecl
  .   "C" classname
  .   FieldDeclList [1]
  .   . (private) FieldDecl
  .   .   BOOLEAN BaseType
  .   .   "b" fieldname
  .   MethodDeclList [1]
  .   . (public) MethodDecl
  .   .   ArrayType
  .   .     INT BaseType
  .   .   "tryit" methodname
  .   .   ParameterDeclList [0]
  .   .   StmtList [2]
  .   .   . VarDeclStmt
  .   .   .   VarDecl
  .   .   .     INT BaseType
  .   .   .     "x" varname
  .   .   .   UnaryExpr
  .   .   .     "-" Operator
  .   .   .       UnaryExpr
  .   .   .         "-" Operator
  .   .   .           RefExpr
  .   .   .             IdRef
  .   .   .               "x" Identifier
  .   .   . ReturnStmt
  .   .   .   NewArrayExpr
  .   .   .     INT BaseType
  .   .   .     LiteralExpr
  .   .   .       "20" IntLiteral
=============================================
